### PR TITLE
Added the Ability to keep the expression

### PR DESCRIPTION
**Pull request by DerDan**
Extends the plugin to keep the arithmetic expression if the expression is followed by an equal sign (=)

`1 + 2 =` after eval  `1 + 2 = 3`

### CODE REVIEW

File: src/main/java/EvaluateArithmeticAction.java [hunk 1]
```diff
@@ -42,10 +42,11 @@ public void actionPerformed(@NotNull final AnActionEvent event) {
      */
     static String evaluate(@NotNull String expression_string) {
         // Only allow arithmetic characters and whitespace
-        if (expression_string.matches("^[0-9+\\-/*^().\\s]*")) {
+        if (expression_string.matches("^[0-9+\\-/*^().\\s]*(=\\s*)?")) {
             try {
+                boolean keep = expression_string.contains("=");
```
Comment (jaredkhan) [inline src/main/java/EvaluateArithmeticAction.java]: `keep` is a little unspecific. How about `should_append_result`?

File: src/main/java/EvaluateArithmeticAction.java [hunk 2]
```diff
@@ -42,10 +42,11 @@ public void actionPerformed(@NotNull final AnActionEvent event) {
...
                 // Normalise all whitespace to spaces. This means groovy expects a single expression
```
Comment (jaredkhan) [inline src/main/java/EvaluateArithmeticAction.java]: Could we also mention the removal of the `=` in the comment? Regex's are a little hard to skim read 😊

File: README.md [hunk 3]
```diff
@@ -3,6 +3,11 @@ A plugin for IntelliJ IDEs. Download [here](https://plugins.jetbrains.com/plugin
 
 ![multi-caret](images/multi_caret.gif)
 
+Possibility to keep the expression if the expression is followed by an equal sign ("=")
+
+![keep_expression](images/keep_expression.gif)
```
Comment (jaredkhan) [inline README.md]: Thanks very much for creating the GIF!

The README doesn't have GIFs for all the features so we probably don't need this one. How about we leave out this README change (but leave the GIF in the MR) and I'll upload the GIF to the Jetbrains page when I make the next release?

File: src/test/java/ArithmeticTests.java [hunk 4]
```diff
@@ -47,6 +47,14 @@ public void precendenceWorks() {
         assertEquals("9", EvaluateArithmeticAction.evaluate("(1 + 2) * 3"));
     }
 
+    @Test
+    public void keepExpression() {
```
Comment (jaredkhan) [inline src/test/java/ArithmeticTests.java]: Worth testing non-space whitespace? (For example: tab/newline)

Comment (jaredkhan) [inline src/test/java/ArithmeticTests.java]: Could you name the test according to the capability it is verifying, potentially starting with `can`? e.g. `canAppendResultToExpression`

**Review by jaredkhan — COMMENTED** [review]
Hey @DerDan! Thanks for taking the time to open this Pull Request. A nice idea, and a nice short implementation 👍 

The feature makes sense, after some thought I don't think it's _too_ surprising that it would keep the expression in place if there is an = in the selection. So yeah, let's do it! I've left a few comments, take a look if you get time 😊 

Cheers

File: src/test/java/ArithmeticTests.java [hunk 5]
```diff
@@ -47,6 +47,14 @@ public void precendenceWorks() {
...
+        assertEquals("1 + 1 =2", EvaluateArithmeticAction.evaluate("1 + 1 ="));
+        // add whitespace after "="
+        assertEquals("5 * 5 = 25", EvaluateArithmeticAction.evaluate("5 * 5 = "));
+    }
+
+
```
Comment (jaredkhan) [inline src/test/java/ArithmeticTests.java:64]: Worth also adding a test that it will ignore things with content to the right of the `=`?

Comment (DerDan) [reply src/test/java/ArithmeticTests.java:64]: Only whitespace should be allowed after the arithmetic expression. Even if the expression is concluded with an equal sign "=". But sure, we also need a test for that

File: src/main/java/EvaluateArithmeticAction.java (hunk 2 above)
Comment (DerDan) [reply src/main/java/EvaluateArithmeticAction.java]: sure

File: src/main/java/EvaluateArithmeticAction.java (hunk 1 above)
Comment (DerDan) [reply src/main/java/EvaluateArithmeticAction.java]: `boolen appendResultToExpression = ` ?

File: README.md (hunk 3 above)
Comment (DerDan) [reply README.md]: is okay for me

File: src/test/java/ArithmeticTests.java (hunk 4 above)
Comment (DerDan) [reply src/test/java/ArithmeticTests.java]: okay

File: src/main/java/EvaluateArithmeticAction.java (hunk 1 above)
Comment (jaredkhan) [reply src/main/java/EvaluateArithmeticAction.java]: Works for me 👍

Comment (DerDan) [reply src/main/java/EvaluateArithmeticAction.java]: done

File: src/main/java/EvaluateArithmeticAction.java (hunk 2 above)
Comment (DerDan) [reply src/main/java/EvaluateArithmeticAction.java]: done

File: src/test/java/ArithmeticTests.java (hunk 4 above)
Comment (DerDan) [reply src/test/java/ArithmeticTests.java]: dona

File: src/test/java/ArithmeticTests.java (hunk 5 above)
Comment (DerDan) [reply src/test/java/ArithmeticTests.java:64]: done

**Comment by jaredkhan** [comment]
Nice work @DerDan. Merging

**Comment by jaredkhan** [comment]
Released https://plugins.jetbrains.com/plugin/13778-evaluate-arithmetic/update/85173

**Pull request merged**